File: 7-kyu/turn-with-a-compass/python/solution.py
def change_dir(dirs, facing, turn):
    curr_dir = dirs.index(facing)
    curr_dir += turn // 45
    return dirs[curr_dir % len(dirs)]

def direction(facing, turn):
    cw_dirs = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
    ccw_dirs = ['N', 'NW', 'W', 'SW', 'S', 'SE', 'E', 'NE']
    return change_dir(cw_dirs, facing, turn) if turn > 0 else change_dir(ccw_dirs, facing, abs(turn))

File: 7-kyu/turn-with-a-compass/javascript/solution.js
const changeDirection = (dirs, facing, turn) => {
  const currentDir = dirs.indexOf(facing) + Math.floor(turn / 45);
  return dirs[currentDir % dirs.length];
};

const direction = (facing, turn) => {
  const clockwise = ["N", "NE", "E", "SE", "S", "SW", "W", "NW"];
  const counterClockwise = ["N", "NW", "W", "SW", "S", "SE", "E", "NE"];
  return turn > 0 ? changeDirection(clockwise, facing, turn) : changeDirection(counterClockwise, facing, Math.abs(turn));
};
